File: KITE-Example-Test/src/main/java/org/webrtc/kite/example/pages/GoogleSearchPage.java
package org.webrtc.kite.example.pages;

import io.cosmosoftware.kite.interfaces.Runner;
import io.cosmosoftware.kite.pages.BasePage;
import org.openqa.selenium.Keys;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class GoogleSearchPage extends BasePage {
  
  private final static String GOOGLE_PAGE = "https://www.youtube.com/watch?v=DCgfVBoBy9s";
  
  @FindBy(className = "gLFyf")
  WebElement searchBar;
  
  
  public GoogleSearchPage(Runner runner) {
    super(runner);
  }
  
  public static String getURL() {
    return GOOGLE_PAGE;
  }
  
  public void open() {
    webDriver.get(GOOGLE_PAGE);
  }
  
  public void searchFor(String target) {
    searchBar.sendKeys(target);
    searchBar.sendKeys(Keys.ENTER);
  }
}

File: KITE-Example-Test/js/pages/google-search-page.js
const {By, Key} = require('selenium-webdriver');
const {TestUtils} = require('kite-common');



const elements = {
  searchBar: By.className('gLFyf'),
}


module.exports = {
  searchFor: async function(stepInfo) {
    let searchBar = await stepInfo.driver.findElement(elements.searchBar);
    await searchBar.sendKeys(stepInfo.target);
    await searchBar.sendKeys(Key.ENTER);
    await TestUtils.waitAround(1000);
  }
}